fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBStoreBase.java

```diff
@@ -41,9 +41,13 @@
 public abstract class FDBStoreBase {
     @Nonnull
     protected final FDBRecordContext context;
+
     @Nonnull
     protected final SubspaceProvider subspaceProvider;
 
+    @Nullable
+    private CompletableFuture<Subspace> subspaceFuture;
```

Comment:
Since this has two stable states, `null` and `completedFuture(Subspace)`, how about just storing a `Subspace` and having `getSubspaceAsync` fill it? Then `getSubspace` can be slightly more efficient, too, using a null check to decide whether to do `asyncToSync` at all.

Reply:
This code is superfluous as there is caching inside SubspaceProviderByKeySpacePath. Should we just remove it?

Reply:
Both recommendations accepted and resolved in the latest push of this branch.